docker/Dockerfile.ci

```diff
@@ -1,4 +1,4 @@
-FROM cimg/python:3.9.10
+FROM cimg/python:3.10.10
```

Comment:
The existing image did not contain the `sqlite3` executable required by the actual dump operation:
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/1894533/229246197-8defb775-20eb-4ca3-9d9b-9c878b052881.png">

Just installing it via `apt` resulted in an older version that we do not support. The [link we provide](https://stackoverflow.com/questions/49920444/upgrading-sqlite3-version-used-in-python3-on-linux/55729735#55729735) with instructions on how to upgrade one's installation is not accurate anymore, requiring more debugging from the user:
```
$ sudo add-apt-repository -y ppa:dqlite/stable
ERROR: ppa 'dqlite/stable' not found (use --login if private)
```

Instead of debugging that, I just switched to a newer image base, as we will be needing it for matrix version local validation in the near future anyway.